sql/src/main/java/io/crate/expression/scalar/string/StringPaddingFunction.java

```diff
@@ -0,0 +1,136 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.expression.scalar.ThreeParametersFunction;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.List;
+import java.util.Locale;
+
+public class StringPaddingFunction extends Scalar<String, Object> {
+
+    public static final String LNAME = "lpad";
+    public static final String RNAME = "rpad";
+    public static final String DEFAULT_FILL = " ";
+    public static final int LEN_LIMIT = 1000;
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(LNAME, new Resolver(LNAME, StringPaddingFunction::lpad));
+        module.register(RNAME, new Resolver(RNAME, StringPaddingFunction::rpad));
+    }
+
+    private final FunctionInfo info;
+    private final ThreeParametersFunction<char[], Integer, char[], String> func;
+
+    private StringPaddingFunction(FunctionInfo info, ThreeParametersFunction<char[], Integer, char[], String> func) {
+        this.info = info;
+        this.func = func;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, Input[] args) {
+        assert args.length == 2 || args.length == 3 : String.format(
+            Locale.ENGLISH,
+            "number of arguments must be 2 (optionally 3), got %d instead",
+            args.length);
+
+        String str = (String) args[0].value();
+        Number len = (Number) args[1].value();
+        String fill = args.length == 3 ? (String) args[2].value() : DEFAULT_FILL;
+        if (str == null || len == null || fill == null) {
+            return null;
+        }
+        int lenValue = len.intValue();
+        if (lenValue > LEN_LIMIT) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                                                             "len argument exceeds predefined limit of %d",
+                                                             LEN_LIMIT));
+        }
+
+        if (lenValue <= 0 || (str.isEmpty() && fill.isEmpty())) {
```

Comment:
Isn't the `fill.isEmpty` case still valid for a truncation? 

```
select lpad('hello', 3, '');
 lpad 
------
 hel
(1 row)
```

Reply:
yes, that works just fine, the test in my if statement is checking the case where both string parameters are the empty string: 
postgres=# select lpad('', 42, '');
 lpad
------

(1 row)